src/prommis/uky/costing/models/tests/test_hd_ce.py

```diff
@@ -0,0 +1,807 @@
+    @pytest.fixture(scope="class")
+    def model(self):
+        model = base_model()
+        return model
+
+    @pytest.mark.component
+    def test_base_model_diagnostics(self, model):
+        dt = DiagnosticsToolbox(model)
+        dt.assert_no_structural_warnings()
+
+    # costing test
+    @pytest.mark.component
+    def test_hd_costing(self, model):
+        model.fs.hydrogen_decrepitation_furnace.costing = UnitModelCostingBlock(
+            flowsheet_costing_block=model.fs.costing,
+            costing_method=REEEquipmentCostingData.cost_hydrogen_decrepitation_furnace,
+            costing_method_arguments={
+                "ramp_up_time": 300,  # (in seconds)
+                "operating_temperature": 443.15,  # in Kelvin
+                "decrepitation_duration": 10800,  # (in seconds)
+                "preparation_time": 3600,  # in seconds
+                "cool_down_time": 3600,  # in seconds
+                "sample_heat_capacity": 0.44,  # in kJ/(kg*K)
+                "sample_mass": None,  # kg
+                "sample_volume": 0.002,  # m**3
+                "sample_density": 7500,  # in kg/m**3
+                "chamber_to_sample_ratio": 2,
+                "length_insulation1": 7.62,  # in m
+                "width_insulation1": 0.6096,  # in m
+                "thickness_insulation1": 0.0254,  # in m
+                "price_insulation1": 183.81,  # in USD
+                "weight_insulation1": 15.42,  # in kg
+                "thermal_cond_insulation_material1": 0.33,  # W/(m*K)
+                "price_metal1": 3.14,  # USD/kg
+                "density_metal1": 7473.57,  # kg/m**3
+                "thermal_cond_metal_material1": 13.53,  # W/(m*K)
+                "length_insulation2": 1.19,  # in m
+                "width_insulation2": 0.381,  # in m
+                "thickness_insulation2": 0.0889,  # in m
+                "price_insulation2": 47.00,  # in USD
+                "weight_insulation2": 10.43,  # in kg
+                "thermal_cond_insulation_material2": 0.069,  # W/(m*K)
+                "price_metal2": 1.50,  # USD/kg
+                "density_metal2": 7861.09,  # kg/m**3
+                "thermal_cond_metal_material2": 45,  # W/(m*K)
+                "hours_per_shift": 8,  # hr
+                "shifts_per_day": 3,
+                "specific_heat_capacity_insulation1": 1.08,  # KJ/(kg*K)
+                "specific_heat_capacity_metal1": 0.468,  # KJ/(kg*K)
+                "specific_heat_capacity_insulation2": 0.9,  # KJ/(kg*K)
+                "specific_heat_capacity_metal2": 0.502416,  # KJ/(kg*K)
+                "operating_days_per_year": 336,  # days
+                "efficiency": 0.95,
+                "electricity_rate": 0.081,  # USD/kWhr
+                "labor_rate": 75,  # USD/hr
+                "temperature_controller_price": 129.00,  # USD
+                "number_of_units": 1,
+                "engineering_and_drafting": 1000,  # USD
+            },
+        )
+
```

Comment:
Add checks on `model.fs.hydrogen_decrepitation_furnace.costing.costing_package.base_currency` and `model.fs.hydrogen_decrepitation_furnace.costing.costing_package.base_period`. Since you set `m.fs.costing = REEEquipmentCosting()`, we expect the units to be `units.USD_Jan_2024` and `units.year`, respectively.